crates/client/src/fuzzer/snapshot_generator.rs

```diff
@@ -412,50 +412,63 @@ fn deserialize_account_tokens(
     deser_method: TokenStream,
 ) -> TokenStream {
     if is_optional {
+        let name_str = name.to_string();
         quote! {
             let #name:Option<#return_type> = accounts_iter
             .next()
-            .ok_or(FuzzingError::NotEnoughAccounts)?
+            .ok_or(FuzzingError::NotEnoughAccounts(#name_str.to_string()))?
             .map(|acc| {
                 if acc.key() != PROGRAM_ID {
-                    #deser_method.map_err(|e| e.to_string())
-                } else {
-                    Err("Optional account not provided".to_string())
+                    #deser_method.map_err(|_| FuzzingError::CannotDeserializeAccount(#name_str.to_string()))
+                } else {Err(FuzzingError::OptionalAccountNotProvided(
+                        #name_str.to_string(),
+                    ))
                 }
             })
             .transpose()
             .unwrap_or(None);
         }
     } else {
+        let name_str = name.to_string();
         quote! {
             let #name: #return_type = accounts_iter
             .next()
-            .ok_or(FuzzingError::NotEnoughAccounts)?
+            .ok_or(FuzzingError::NotEnoughAccounts(#name_str.to_string()))?
             .map(|acc| #deser_method)
-            .ok_or(FuzzingError::AccountNotFound)?
-            .map_err(|_| FuzzingError::CannotDeserializeAccount)?;
+            .ok_or(FuzzingError::AccountNotFound(#name_str.to_string()))?
+            // TODO It would be helpful to do something like line below.
```

Comment:
I tried to resolve it but without success. It really looks like some issue due to reexporting the type and having different anchor versions. Not sure. For now it is not critical. I will try to look at it once we update to anchor 0.29.0....